Include gravity resources in debug report

bernardjkim opened the pull request:
## Description
<!--Required. Provide high-level overview of what the change is for.-->
This PR extends the gravity debug report to include the gravity resources in yaml format.

Contents of the debug report after extracting resources:
```
[vagrant] tree report 
report
├── cluster.json
├── node-1-debug-logs.tar.gz
├── node-1-etcd.tar.gz
├── node-2-debug-logs.tar.gz
├── node-2-etcd.tar.gz
├── node-3-debug-logs.tar.gz
├── node-3-etcd.tar.gz
├── node-3-k8s-logs.tar.gz
├── node-3-resources
│   ├── alert.yaml
│   ├── authgateway.yaml
│   ├── clusterconfiguration.yaml
│   ├── logforwarder.yaml
│   ├── operations.yaml
│   └── runtimeenvironment.yaml
├── node-3-resources.tar.gz
├── node-3-status.tar.gz
└── operation_install.a27a9fa0-c5e6-4a62-973b-2893752c25b6
```

Journal logs are now collected as a regular text file.

## Type of change
<!--Required. Keep only those that apply.-->

* New feature (non-breaking change which adds functionality)

## Linked tickets and other PRs
<!--Required. Keep only those that apply.-->

<!--This PR addresses the following issues.-->
* Refs https://github.com/gravitational/gravity/issues/2139, https://github.com/gravitational/gravity/issues/1978

## TODOs
<!--Required. Keep only those that apply and check them off as they get completed.-->

- [x] Self-review the change
- [x] Perform manual testing
- [ ] Address review feedback

## Testing done
<!--Required. Explain what kind of testing these changes underwent.-->
**Verify `gravity report` command includes `<node>-resources.tar.gz`**

**Verify `gravity-journal.log` and `planet-journal.log` are formatted as a normal text file.**

bernardjkim committed:
Include gravity resources to debug report

bernardjkim committed:
Collect journal logs as a regular text file

a-palchikov commented on lib/report/resources.go:
```diff
@@ -0,0 +1,51 @@
+	// Collect select gravity resources. More information on supported gravity
+	// resources found at https://gravitational.com/gravity/docs/config/
+	resources := []string{
+		"clusterconfiguration",
```
Please use the predefined constants: https://github.com/gravitational/gravity/blob/version/7.0.x/lib/storage/resources.go#L53-L79

a-palchikov commented on lib/report/resources.go:
```diff
@@ -0,0 +1,51 @@
+// gravityResourceYAML returns the gravity command to output the specified
+// resource in YAML format.
+func gravityResourceYAML(resource string) []string {
+	return []string{utils.Exe.Path, "resource", "get", "--format", "yaml", resource}
```
```go
utils.Self("resource", "get", "--format", "yaml", resource)
```

a-palchikov commented on lib/report/system.go:
```diff
@@ -151,11 +151,7 @@ cat %v 2> /dev/null || true`
 // planetLogs fetches planet syslog messages as well as the fresh journal entries
 func planetLogs(since time.Duration) Collectors {
 	return Collectors{
-		// Fetch planet journal entries for the last two days
```
This will effectively prevent recovering structured journal logs - I thought the agreement was that we **add** plain-text logs in addition to structured ones - @r0mant @bernardjkim?

bernardjkim replied:
Oh, we can include both formats. We'll keep the `planet-journal-export.log` and add a plain text `planet-journal.log`. We'll do the same for the gravity journal logs as well.

bernardjkim committed:
Address feedback

- Use storage pkg constants.
- Use utils.Self.
- Collect journal log in both plain and exported format.
